### PR TITLE
[CAS-532] Add different channel item layout for shared groups screen

### PR DESCRIPTION
**Pull request by adasiewiczr**
Screenshot 2021-01-13 at 14 46 59

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt [hunk 1]
```diff
@@ -158,8 +162,8 @@ public class ChannelListView @JvmOverloads constructor(
 
     public override fun onVisibilityChanged(view: View, visibility: Int) {
         super.onVisibilityChanged(view, visibility)
-        if (visibility == View.VISIBLE) {
-            requireAdapter().notifyDataSetChanged()
+        if (visibility == View.VISIBLE && ::adapter.isInitialized) {
+            adapter.notifyDataSetChanged()
```
Comment (adasiewiczr) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:166]: `requireAdapter()` initializes the adapter when it is not initialized yet and we cannot set custom view holder if the adapter is already initialized.

Comment (carterhudson) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:166]: I guess this implementation changed recently. The way I had it configured previously, you could set a VHF at any point.

**Review by carterhudson — APPROVED** [review]
Looks good. You might have some merge conflicts with colors if Marton's PR gets merged first.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt (hunk 1 above)
Comment (zsmb13) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/ChannelListView.kt:166]: Yes, sorry, this was me in https://github.com/GetStream/stream-chat-android/pull/1163, where I change the channel list API to match how the message list API works

**Review by zsmb13 — APPROVED** [review]
LGTM

**Pull request merged**